lib/tasks.js

```diff
@@ -587,4 +587,58 @@ module.exports = function(self, options) {
     await resolveDeferredRelationships();
     console.log('... Done.');
   };
+
+  self.enableCleanPagesTreeTask = function() {
+    self.addTask('clean-pages-tree',
+      `Cleans your pages tree in the case the live one differs with the draft one,
+      The live tree will be modified based on the draft one.`,
+      self.cleanPagesTree
+    );
+  };
+
+  self.cleanPagesTree = async function() {
+    const locales = Object.keys(self.apos.workflow.locales)
+      .filter((locale) => !locale.includes('-draft'));
+
+    const pageTypes = self.apos.modules['apostrophe-pages'].options.types.map((type) => type.name);
+
+    for (const locale of locales) {
+      const pages = await self.apos.docs.db.find({
+        workflowLocale: {$in: [locale, self.draftify(locale)]},
+        type: { $in: pageTypes }
+      }).toArray();
+
+      // We group pages by workflowGuid
+      const groupedPages = pages.reduce((acc, page) => {
+        const isDraft = page.workflowLocale.includes('-draft');
+
+        return {
+          ...acc,
+          [page.workflowGuid]: {
+            ...acc[page.workflowGuid] && acc[page.workflowGuid],
+            ...isDraft ? { draft: page } : { live: page }
+          }
+        };
+
+      }, {});
+
+      for (const { draft, live } of Object.values(groupedPages)) {
+        try {
+          await self.apos.docs.db.update({
```

Comment:
Although it should not happen, it would be best if this did not crash when the live, or the draft, does not exist somehow. It does not have to fix that situation, but it should be able to continue.

Reply:
Condition added.